trademapper/js/trademapper.route.js

```diff
@@ -106,8 +171,74 @@ define([], function() {
 				}
 			}
 		}
+
+		this.collapsePoints();
 	}
 
+	// If there are multiple points with the same locationGroupId in the
+	// points array, use the point with the highest precedence
+	// for that role (latlong > port > country code).
```

Comment:
Maybe I missed it, but I didn't see anything that tests this logic. Maybe it would be worthwhile to have a test for it?

Reply:
It's tested here:

https://github.com/trademapper/trademapper-js/blob/elliots-ports/tests/js/test-route.js#L125

But might be a good idea to check the port > country precedence in the same test, too.